### ISSUE TITLE
Add ACL abuse to impacket/examples/ldap_shell.py

### ISSUE THREAD

**Issue by z-Riocool** [SecureAuthCorp/impacket#1100]
Hi, On June 17 an article came out demonstrating a way to get a TGT ticket or NTLM hash via the msDS-KeyCredentialLink attribute. 
Post: https://posts.specterops.io/shadow-credentials-abusing-key-trust-account-mapping-for-takeover-8ee1a53566ab 
C# tool: https://github.com/eladshamir/Whisker
It would be great if this abuse functionality could be added to ldap_shell.

**Comment by ShutdownRepo** [comment]
Hey @z-Riocool, obtaining the NT hash of a user through a specific TGS request can be made after a PKINIT pre-authentication (as far as my understanding goes). There are now tools that allow to do this things from UNIX and from relayed authentications.

- [PKINITtools](https://github.com/dirkjanm/PKINITtools) is a set of tools allowing users to obtain TGTs through PKINIT and obtain the NT hash as requested in your original post. To do this, you need knowledge of a certificate. The PKINIT pre-authentication relies on this.
- [pyWhisker](https://github.com/ShutdownRepo/pywhisker): is a Python equivalent of Whisker. You can operate a Shadow Credentials attack with it.
- Pull Request #1132 is waiting approval. It features pyWhisker's "add" feature to allow attackers to "add a certificate to a user" through a relayed authentication.

In conclusion, the feature you are requesting is now awaiting approval (#1132) :smile:
